http-client/src/main/java/io/airlift/http/client/jetty/JettyHttpClient.java

```diff
@@ -672,7 +674,19 @@ public <T, E extends Exception> T execute(Request request, ResponseHandler<T, E>
         }
     }
 
-    public <T, E extends Exception> T doExecute(Request request, ResponseHandler<T, E> responseHandler, Span span)
+    @Override
+    public StreamingResponse executeStreaming(Request request)
+    {
+        request = applyRequestFilters(request);
+
+        Span span = startSpan(request);
+        request = injectTracing(request, span);
+
+        StreamingResponseHandler responseHandler = new StreamingResponseHandler(span);
+        return doExecute(request, responseHandler, span, responseHandler);
```

Comment:
Do we need this StreamingResponseHandler? Can we do it without it?

Reply:
I guess we could inline it in `executeStreaming`. Would you prefer that?

Reply:
Yes. We wanted to avoid using that abstraction for this purpose as it doesn't fit

Reply:
@wendigo reworked the code - PTAL